src/remoteGrading.ts

```diff
@@ -29,8 +29,21 @@ export async function doRemoteGrading(
       REQUEST_TIMEOUT_MS,
     );
 
+    logger.debug(
+      `Got remote grading result: status=${status}, statusText=${statusText}, data=${JSON.stringify(data)}`,
+    );
+
+    if (status !== 200) {
+      throw new Error(
+        `Remote grading failed with status ${status}: ${statusText} ${JSON.stringify(data)}`,
+      );
+    }
     const { result } = data as { result: GradingResult };
-    logger.debug(`Got remote grading result: ${JSON.stringify(result)}`);
+
+    if (!result) {
```

Comment:
if you're doing this you might as well consider checking if pass exists (only critical field)